source/common/router/config_impl.cc

```diff
@@ -401,7 +404,13 @@ std::string RouteEntryImplBase::newPath(const Http::HeaderMap& headers) const {
     final_path = path_redirect_.c_str();
   } else {
     ASSERT(headers.Path());
-    final_path = headers.Path()->value().c_str();
+    final_path = absl::string_view(headers.Path()->value().c_str());
```

Comment:
Sorry, forgot that the header value isn't a std::string. This ends up calling`strlen`. Since we know the length already via `headers.Path()->value().size()` you can all the `absl::string_view(const char*, size_t)` constructor to avoid the strlen.